File: backend/app/schemas/category.py
from typing import Optional, List
import uuid
from pydantic import BaseModel, ConfigDict

class CategoryBase(BaseModel):
    name: str
    description: Optional[str] = None

class CategoryCreate(CategoryBase):
    pass

class CategoryUpdate(CategoryBase):
    name: Optional[str] = None
    description: Optional[str] = None

class CategoryResponse(CategoryBase):
    id: uuid.UUID
    plant_count: int = 0

    model_config = ConfigDict(from_attributes=True)

class CategoryListResponse(BaseModel):
    items: List[CategoryResponse]
    total: int

File: frontend/src/types/category.ts
export interface Category {
    id: string;
    name: string;
    description?: string;
    plant_count?: number;
}

export type CategoryResponse = Category;

export interface CategoryCreate {
    name: string;
    description?: string;
}

export interface CategoryUpdate {
    name?: string;
    description?: string;
}

export interface CategoryListResponse {
    items: CategoryResponse[];
    total: number;
}
